### PR TITLE
Fixes #18104 : change parse_obj and assertEquals which was deprecated

### LINKED ISSUE #18104: python test warning (deprecate and so on)
**Affected module**
When we run python test, many wanrining occured.
so, I should fix them little by little.

**Describe the bug**
parse_obj -> model_validate
assertEquals -> assertEqual
dict -> model_dump

**To Reproduce**

just replace

**Expected behavior**
nothing

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### PR DESCRIPTION
**Pull request by kuma0128**
### Describe your changes:

Fixes #18104

I worked on ... because ...

#
### Type of change:

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

**Comment by pmbrull** [comment]
thanks for the PR! enabled the CI, let's see if it likes everything

**Comment by pmbrull** [comment]
hi aside from the formatting, not sure if you were able to test the backend changes on your end. Looks like CI cannot run

```
Error:  /home/runner/work/OpenMetadata/OpenMetadata/openmetadata-service/src/test/java/org/openmetadata/service/util/TestUtils.java:[539,7] cannot find symbol
  symbol:   method assertEqual(int,int)
  location: class org.openmetadata.service.util.TestUtils
```

Python changes LGTM though

**Comment by kuma0128** [comment]
The assertEquals in  python and the assertEquals in Junit were actually different things. I need to correct this.

File: openmetadata-service/src/test/java/org/openmetadata/service/security/policyevaluator/PolicyEvaluatorTest.java [hunk 1]
```diff
@@ -94,8 +94,8 @@ void trimResourcePermissions() {
 
     List<ResourcePermission> rpList = List.of(rp1, rp2);
     PolicyEvaluator.trimResourcePermissions(rpList);
-    assertEqualsPermissions(expectedOp1, rpList.get(0).getPermissions());
-    assertEqualsPermissions(expectedOp2, rpList.get(1).getPermissions());
+    assertEqualPermissions(expectedOp1, rpList.get(0).getPermissions());
+    assertEqualPermissions(expectedOp2, rpList.get(1).getPermissions());
```
Comment (pmbrull) [inline openmetadata-service/src/test/java/org/openmetadata/service/security/policyevaluator/PolicyEvaluatorTest.java:98]: should we revert this change just to keep the PR clean? Seems like a " replace all " kicked in

Comment (kuma0128) [reply openmetadata-service/src/test/java/org/openmetadata/service/security/policyevaluator/PolicyEvaluatorTest.java:98]: I will fix and check

Comment (kuma0128) [reply openmetadata-service/src/test/java/org/openmetadata/service/security/policyevaluator/PolicyEvaluatorTest.java:98]: In java, "assertEquials" is correct, so I revert this change.

**Review by pmbrull — APPROVED** [review]
Thanks for the changes, LGTM

**Pull request merged**